samtranslator/model/sam_resources.py

```diff
@@ -2590,5 +2625,190 @@ def make_runtime_dict(r: aws_serverless_graphqlapi.Runtime) -> AppSyncRuntimeTyp
 
         # Runtime is not defined, raise error.
         raise InvalidResourceException(
-            relative_id, "'Runtime' must be defined as a property here or in 'ResolverCodeSettings.'"
+            relative_id, f"'Runtime' must be defined as a property in {relative_id} or in 'ResolverCodeSettings.'"
+        )
+
+    def _construct_appsync_resolver_resources(
+        self,
+        appsync_resolvers: Dict[str, Dict[str, aws_serverless_graphqlapi.AppSyncResolver]],
+        resolver_code_settings: Optional[aws_serverless_graphqlapi.ResolverCodeSettings],
+        api_id: Intrinsicable[str],
+    ) -> List[Resource]:
+        resources: List[Resource] = []
+
+        # Because resolvers are stored in a unique manner with the parent TypeName dictionary,
+        # we must first gather all relative id to AppSyncResolver pairs and put them into one dictionary
+        # before passing it to _check_and_construct_none_datasource() which expects a dictionary in format
+        # {logical_id: resource}.
+        resolvers = {}
+        for resolver in appsync_resolvers.values():
+            resolvers.update(resolver)
+
+        self._check_and_construct_none_datasource(resolvers, api_id)
+
+        for type_name, relative_id_to_resolver in appsync_resolvers.items():
```

Comment:
(nit): `relative_id_and_resolver` or `other_resolver_properties`

Reply:
Why `relative_id_and_resolver`? It's a dictionary, naming it `<key>_to_<value>`should be fine?